crates/biome_diagnostics/src/adapters.rs

```diff
@@ -156,3 +160,39 @@ impl Diagnostic for SerdeJsonError {
         fmt.write_markup(markup!({ AsConsoleDisplay(&self.error) }))
     }
 }
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct IniError {
+    pub message: String,
+
+    #[serde(skip)]
+    pub source: Option<serde_ini::de::Error>,
+}
```

Comment:
An adapter should only contain an external error, nothing else. It's just a wrapper that knows how to print the external error. Instead, the message should be set in the ` EditorConfigValidationError::ParseFailed` diagnostic.

So you'll have something like this (also, follow the same code structure of the current errors, no need to deviate):

```rs
// biome_diagnostics/adapters.rs
#[derive(Debug)]
struct IniError {
	error: Option<serde_ini::de::Error>,
}

// biome_service/diagnostics.rs
enum EditorConfigDiagnostic {
	ParseFailed(ParseFailedDiagnostic)
}

#[derive(Diagnostic)]
#[diagnostic(
	message = "Failed the parse the .editorconfig file."
)]
enum ParseFailedDiagnostic {
		#[source]
		source: IniError
}

fn parse() {
	parse.map_err(err|| {
		EditorConfigDiagnostic::ParseFailed(ParseFailedDiagnostic {
			source: Error::from(IniError::from(err))
		})
	})
}
```